src/borg/helpers/fs.py

```diff
@@ -286,6 +286,13 @@ def os_open(*, flags, path=None, parent_fd=None, name=None, noatime=False):
                 raise
             # Was this EPERM due to the O_NOATIME flag? Try again without it:
             fd = os.open(fname, _flags_normal, dir_fd=parent_fd)
+        except OSError as exc:
+            # O_NOATIME causes EROFS when accessing a volume shadow copy in WSL1
+            from . import workarounds
```

Comment:
did you try to do this import on global level instead of here?

Reply:
Yes in that case import fails due to circular import.

https://github.com/borgbackup/borg/blob/bf392367f159fd81494a9d76bb646e7f3ab37132/src/borg/helpers/__init__.py#L12

Reply:
ok, i had a similar effect recently, too.